src/openrct2/ride/CableLift.cpp

```diff
@@ -119,104 +113,104 @@ void Vehicle::CableLiftUpdate()
+    velocity = 0;
+    acceleration = 0;
+    SetState(VEHICLE_STATUS_WAITING_FOR_PASSENGERS, sub_state);
 }
 
 /**
  *
  *  rct2: 0x006DF8F1
  */
-static void cable_lift_update_waiting_to_depart(Vehicle* vehicle)
+void Vehicle::CableLiftUpdateWaitingToDepart()
 {
-    if (vehicle->velocity >= -58640)
-        vehicle->acceleration = -14660;
+    if (velocity >= -58640)
+        acceleration = -14660;
 
-    if (vehicle->velocity < -58640)
+    if (velocity < -58640)
     {
-        vehicle->velocity -= vehicle->velocity / 16;
-        vehicle->acceleration = 0;
+        velocity -= velocity / 16;
+        acceleration = 0;
     }
 
-    cable_lift_update_track_motion(vehicle);
+    cable_lift_update_track_motion(this);
 
     // Next check to see if the second part of the cable lift
     // is at the front of the passenger vehicle to simulate the
     // cable being attached underneath the train.
-    Vehicle* passengerVehicle = GET_VEHICLE(vehicle->cable_lift_target);
-    Vehicle* cableLiftSecondPart = GET_VEHICLE(vehicle->prev_vehicle_on_ride);
+    Vehicle* passengerVehicle = GET_VEHICLE(cable_lift_target);
+    Vehicle* cableLiftSecondPart = GET_VEHICLE(prev_vehicle_on_ride);
 
     int16_t dist_x = abs(passengerVehicle->x - cableLiftSecondPart->x);
     int16_t dist_y = abs(passengerVehicle->y - cableLiftSecondPart->y);
 
     if (dist_x + dist_y > 2)
         return;
 
-    vehicle->velocity = 0;
-    vehicle->acceleration = 0;
-    vehicle->SetState(VEHICLE_STATUS_DEPARTING, 0);
+    velocity = 0;
+    acceleration = 0;
+    SetState(VEHICLE_STATUS_DEPARTING, 0);
 }
 
 /**
  *
  *  rct2: 0x006DF97A
  */
-static void cable_lift_update_departing(Vehicle* vehicle)
+void Vehicle::CableLiftUpdateDeparting()
 {
-    vehicle->sub_state++;
-    if (vehicle->sub_state < 16)
+    sub_state++;
+    if (sub_state < 16)
         return;
 
-    Vehicle* passengerVehicle = GET_VEHICLE(vehicle->cable_lift_target);
-    vehicle->SetState(VEHICLE_STATUS_TRAVELLING, vehicle->sub_state);
+    Vehicle* passengerVehicle = GET_VEHICLE(cable_lift_target);
+    SetState(VEHICLE_STATUS_TRAVELLING, sub_state);
     passengerVehicle->SetState(VEHICLE_STATUS_TRAVELLING_CABLE_LIFT, passengerVehicle->sub_state);
 }
 
 /**
  *
  *  rct2: 0x006DF99C
  */
-static void cable_lift_update_travelling(Vehicle* vehicle)
+void Vehicle::CableLiftUpdateTravelling()
 {
-    Vehicle* passengerVehicle = GET_VEHICLE(vehicle->cable_lift_target);
+    Vehicle* passengerVehicle = GET_VEHICLE(cable_lift_target);
 
-    vehicle->velocity = std::min(passengerVehicle->velocity, 439800);
-    vehicle->acceleration = 0;
+    velocity = std::min(passengerVehicle->velocity, 439800);
+    acceleration = 0;
     if (passengerVehicle->update_flags & VEHICLE_UPDATE_FLAG_BROKEN_TRAIN)
         return;
 
-    if (!(cable_lift_update_track_motion(vehicle) & (1 << 1)))
+    if (!(cable_lift_update_track_motion(this) & (1 << 1)))
```

Comment:
`(1<<1)` in this instance is `VEHICLE_UPDATE_MOTION_TRACK_FLAG_1` could you make this change as well.